[KOGITO-3922] Adding error handling to SWF (#3851)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/serverless/chap-kogito-orchestrating-serverless.adoc

* `functions`: Function definitions are reusable constructs that you can use to define invocation information about services that need to be invoked during workflow execution. Function definitions can be referenced by workflow `states` (control-logic blocks) whenever a service needs to be invoked.
* `events`: Event definitions are reusable constructs that you can use to define all consumed and produced events during workflow execution. Serverless Workflow requires https://cloudevents.io/[CloudEvents] format for both consumed and produced events. Event definitions can be referenced by workflow states to determine how a workflow instance is created, what events are needed to continue workflow execution, what events need to be produced during workflow execution, and how multiple events are associated with one another (correlation rules). You can also use event definitions to trigger function or service invocation for services that do not use exposed invocation endpoints and that are only invoked by events.

> * `states`: State definitions enable you to define your workflow states (control-logic blocks) and determine what your workflow should do. State definitions specify details such as how the workflow is started, what data is expected, how the workflow should transition from one state to another given some data or event constraints, error handling and how and when the workflow execution completes.

Reviewer: ```suggestion
* `states`: Using state definitions, you can define your workflow states (control-logic blocks) and determine what your workflow should do. State definitions specify details such as:

- How the workflow starts.
- The expected data.
- How the workflow transitions from one state to another, given some data or event constraints.
- Error handling.
- How and when the workflow execution completes.
```

Style Guide: Anthromorphism - _Do not assign human characteristics to inanimate objects_. I rewrote to remove "enables" and remove passive voice.

Reviewer 2: Can't we add a new section "Error Handling" and explain how to use it based on our implementation? We now support the `code: <java exception>`. We must have a section explaining how to use it.

> * `states`: State definitions enable you to define your workflow states (control-logic blocks) and determine what your workflow should do. State definitions specify details such as how the workflow is started, what data is expected, how the workflow should transition from one state to another given some data or event constraints, error handling and how and when the workflow execution completes.

Reviewer 3: ```suggestion
* `states`: State definitions enable you to define your workflow states (control-logic blocks) and determine what your workflow should do. State definitions specify details such as how the workflow is started, what data is expected, how the workflow should transition from one state to another given some data or event constraints, error handling, and how and when the workflow execution completes.
```

ifdef::parent-context[:context: {parent-context}]
ifndef::parent-context[:!context:]

> == Serverless Workflow Error handling 

Reviewer 3: ```suggestion
[id="con-serverless-workflow-error-handling_{context}"]
== Error handling in Serverless Workflow
```

> You can handle exceptions thrown by ``operation`` nodes, as described [here](https://github.com/serverlessworkflow/specification/blob/main/specification.md#Workflow-Error-Handling).

Reviewer 3: ```suggestion
You can handle the exceptions that are thrown by `operation` nodes as described in the https://github.com/serverlessworkflow/specification/blob/main/specification.md#Workflow-Error-Handling[Serverless Workflow Specification].
```

> The only consideration specific to Kogito is that, since it is implemented in java,  you need to specify the fully qualified name of the exception that might be thrown by the operation in the ``code`` field of the error definition.

Reviewer 3: ```suggestion
NOTE: You need to specify the qualified name of the exception that might be thrown by the `operation` in the `code` field of the error definition.
```

> Below it is a simple example which invoke an ``EvenService`` which throws ``IllegalArgumentException`` if the number in the workflow model is odd. 

Reviewer 3: ```suggestion
The following is an example that invokes an `EvenService`, which is throwing `IllegalArgumentException` if the number in the workflow model is odd. 
```
